### PR TITLE
Call Start endpoint to start simulated device

### PR DESCRIPTION
**Pull request by ajaxjiang96**

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package main
+
```
Comment (qs-wang) [inline utils.go]: could we change the file name to ping, or network, or whatever? Generally I don't like a "utils" file which contains all unrelated util funcs.

File: utils.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		_, err := client.R().
+			Get(pingUrl)
+		if err != nil {
+			logger.Info("Ping simulator failed", zap.Error(err))
+			time.Sleep(time.Second * time.Duration(sleep))
+			sleep = sleep * 2
+		} else {
+			logger.Info("Simulator device online!")
+			pong = true
+		}
```
Comment (qs-wang) [inline utils.go]: shall we return pong here?

File: activities.go [hunk 3]
```diff
@@ -31,3 +36,29 @@ func runSimulationActivity(ctx context.Context,
...
+func startDeviceActivity(ctx context.Context, port int, deviceJsonbytes []byte) ([]byte, error) {
+	logger := activity.GetLogger(ctx)
+	logger.Info("Starting Simulated Device")
+	client := resty.New()
+
+	pingUrl := fmt.Sprintf("%s:%d/start", os.Getenv("CONTAINER_HOST"), port)
+
+	url := fmt.Sprintf("%s:%d/start", os.Getenv("CONTAINER_HOST"), port)
+
+	PingSimulator(client, pingUrl, logger)
```
Comment (qs-wang) [inline activities.go]: check ping result?

File: activities.go [hunk 4]
```diff
@@ -31,3 +36,29 @@ func runSimulationActivity(ctx context.Context,
...
 }
+
+func startDeviceActivity(ctx context.Context, port int, deviceJsonbytes []byte) ([]byte, error) {
+	logger := activity.GetLogger(ctx)
+	logger.Info("Starting Simulated Device")
+	client := resty.New()
+
+	pingUrl := fmt.Sprintf("%s:%d/start", os.Getenv("CONTAINER_HOST"), port)
+
+	url := fmt.Sprintf("%s:%d/start", os.Getenv("CONTAINER_HOST"), port)
```
Comment (qs-wang) [inline activities.go]: move this url to the line which is closer to the usage.

File: utils.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"go.uber.org/zap"
+)
+
+func PingSimulator(client *resty.Client, pingUrl string, logger *zap.Logger) {
+	pong := false
+	sleep := 1
+	for !pong {
```
Comment (qs-wang) [inline utils.go]: put a timeout limitation here. We probably don't  want to loop infinitely.

File: utils.go (hunk 2 above)
Comment (ajaxjiang96) [reply utils.go]: returning `pong, err` where err is a customized error informing couldn't ping device within timeout durations

File: activities.go (hunk 3 above)
Comment (ajaxjiang96) [reply activities.go]: done

**Pull request merged**